src/Compilers/CSharp/Test/Semantic/Semantics/IteratorTests.cs

```diff
@@ -61,6 +62,67 @@ IEnumerable<int> I()
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void BasicIterators_Async()
+        {
+            var source = """
+                using System.Collections.Generic;
+                using System.Threading.Tasks;
+                
+                class Test
+                {
+                    async IAsyncEnumerable<int> I()
+                    {
+                        await Task.Yield();
+                        yield return 1;
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net60);
+            comp.VerifyDiagnostics();
+
+            var i = comp.GetMember<MethodSymbol>("Test.I");
+            Assert.True(i.IsIterator);
```

Comment:
>Assert.True(i.IsIterator); [](http://example.com/codeflow?start=12&length=26)

It doesn't look like this tests the interface implementation behavior. Probably other tests too.